dlc/src/util.rs

```diff
@@ -0,0 +1,72 @@
+//! Utility functions not uniquely related to DLC
+
+use bitcoin::util::bip143::SighashComponents;
+use bitcoin::{PrivateKey, Script, SigHashType, Transaction};
+use secp256k1::{Message, Secp256k1, Signature, Signing};
+
+/// Get a signature hash for a segwit transaction input as a Message instance
+pub(crate) fn get_sig_hash_msg(
+    tx: &Transaction,
+    input_index: usize,
+    script_pubkey: &Script,
+    value: u64,
+) -> Message {
+    let sig_hash =
+        SighashComponents::new(tx).sighash_all(&tx.input[input_index], script_pubkey, value);
+    Message::from_slice(&sig_hash).unwrap()
+}
+
+/// Convert a raw signature to DER encoded and append the sighash type, to use
+/// a signature in a signature script
+pub(crate) fn sig_to_full_sig(sig: &Signature, sig_hash_type: SigHashType) -> Vec<u8> {
```

Comment:
`finalize_sig`. If we don't have this in rust-bitcoin, maybe add it ?

Reply:
Renamed. I checked in rust-bitcoin but couldn't find something similar, where do you think it would be nice to have? In util/misc.rs?

Reply:
There is a `Signature::serialize_der()` so I don't think if it's that much useful after thinking more, only if you're finalizing a big multisig ? Don't bother for now.